send.php

```diff
@@ -29,64 +29,73 @@
 
 $dir = dirname($file);
 $old = $file;
-$file = $dir . '/analytics-' . rand() . '.log';
 
-if(!file_exists($old)) {
-  print("file: $old does not exist");
-  exit(0);
-}
+$newFileName = 'analytics-' . rand() . '.log';
+$newFile = $dir . '/' . $newFileName;
 
-if (!rename($old, $file)) {
-  print("error renaming from $old to $file\n");
-  exit(1);
+if(file_exists($old)) {
+    if (!rename($old, $newFile)) {
+        print("error renaming from $old to $newFile\n");
+        exit(1);
+    }
 }
 
-/**
- * File contents.
- */
-
-$contents = file_get_contents($file);
-$lines = explode("\n", $contents);
-
-/**
- * Initialize the client.
- */
-
-Segment::init($args["secret"], array(
-  "debug" => true,
-  "error_handler" => function($code, $msg){
-    print("$code: $msg\n");
-    exit(1);
-  }
-));
-
-/**
- * Payloads
- */
-
-$total = 0;
-$successful = 0;
-foreach ($lines as $line) {
-  if (!trim($line)) continue;
-  $payload = json_decode($line, true);
-  $dt = new DateTime($payload["timestamp"]);
-  $ts = floatval($dt->getTimestamp() . "." . $dt->format("u"));
-  $payload["timestamp"] = $ts;
-  $type = $payload["type"];
-  $ret = call_user_func_array(array("Segment", $type), array($payload));
-  if ($ret) $successful++;
-  $total++;
-  if ($total % 100 === 0) Segment::flush();
+foreach(scandir($dir) as $file) {
+    if (substr($file, 0, strlen('analytics-')) !== 'analytics-') {
+        continue;
+    }
+    if ($file === $newFileName) {
+        continue;
+    }
+
+    /**
+    * File contents.
+    */
+
+    $contents = file_get_contents($file);
+    $lines = explode("\n", $contents);
+
+    /**
+    * Initialize the client.
+    */
+
+    Segment::init($args["secret"], array(
```

Comment:
Might want to move Segment initialisation out of the loop since we might end up doing it multiple times.